### PR TITLE
Issue #75 Create PassportDao

### LINKED ISSUE #75: Create PassportDao

### PR DESCRIPTION
**Pull request by ChirkinaTanya**

### CODE REVIEW

**Review by JustRevolt — APPROVED** [review]

File: src/main/java/com/ifmo/epampractice/dao/PassportDao.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    @Override
+    void update(PassportEntity PassportEntity);
+
+    @Override
+    void delete(PassportEntity PassportEntity);
+
+    @Override
+    void save(PassportEntity PassportEntity);
+}
```
Comment (a-deriabin) [inline src/main/java/com/ifmo/epampractice/dao/PassportDao.java]: missing newline

File: src/main/java/com/ifmo/epampractice/daoimpl/PassportDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+import java.util.List;
+
+public class PassportDaoImpl implements PassportDao {
+    private static final String GET_ALL_QUERY = "SELECT * FROM passport";
+    private static final String GET_QUERY = "SELECT * FROM passport WHERE id = ?";
+    private static final String UPDATE_QUERY = "UPDATE passport SET issue_country = ?, issuer = ?, issue_date = ?, expiration_date = ?, serial_number = ? WHERE id = ?";
+    private static final String DELETE_QUERY = "DELETE FROM passport WHERE id = ?";
+    private static final String SAVE_QUERY = "INSERT INTO passport (issue_country, issuer, issue_date, expiration_date, serial_number) VALUES (?, ?, ?, ?, ?)";
+
+    private static final Logger LOGGER = LogManager.getLogger(CarDaoImpl.class);
```
Comment (a-deriabin) [inline src/main/java/com/ifmo/epampractice/daoimpl/PassportDaoImpl.java]: Every constant should be static final, but not every static final is a constant.
In this case, it's just a reference to a logger object.
(UPD: wait with fixing this, maybe it's a part of EPAM codestyle, waiting for mentors' reply)
(UPD2: yes, better make in non-upper case)

File: src/main/java/com/ifmo/epampractice/daoimpl/PassportDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+import org.apache.logging.log4j.Logger;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PassportDaoImpl implements PassportDao {
+    private static final String GET_ALL_QUERY = "SELECT * FROM passport";
+    private static final String GET_QUERY = "SELECT * FROM passport WHERE id = ?";
+    private static final String UPDATE_QUERY = "UPDATE passport SET issue_country = ?, issuer = ?, issue_date = ?, expiration_date = ?, serial_number = ? WHERE id = ?";
```
Comment (a-deriabin) [inline src/main/java/com/ifmo/epampractice/daoimpl/PassportDaoImpl.java]: long line

File: src/main/java/com/ifmo/epampractice/daoimpl/PassportDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    private static final String DELETE_QUERY = "DELETE FROM passport WHERE id = ?";
+    private static final String SAVE_QUERY = "INSERT INTO passport (issue_country, issuer, issue_date, expiration_date, serial_number) VALUES (?, ?, ?, ?, ?)";
```
Comment (a-deriabin) [inline src/main/java/com/ifmo/epampractice/daoimpl/PassportDaoImpl.java]: long line

**Review by nikaalin — APPROVED** [review]

**Review by a-deriabin — CHANGES_REQUESTED** [review]

File: src/main/java/com/ifmo/epampractice/daoimpl/PassportDaoImpl.java (hunk 2 above)
Comment (ChirkinaTanya) [reply src/main/java/com/ifmo/epampractice/daoimpl/PassportDaoImpl.java]: Fixed

File: src/main/java/com/ifmo/epampractice/daoimpl/PassportDaoImpl.java (hunk 3 above)
Comment (ChirkinaTanya) [reply src/main/java/com/ifmo/epampractice/daoimpl/PassportDaoImpl.java]: Fixed

File: src/main/java/com/ifmo/epampractice/daoimpl/PassportDaoImpl.java (hunk 4 above)
Comment (ChirkinaTanya) [reply src/main/java/com/ifmo/epampractice/daoimpl/PassportDaoImpl.java]: Fixed

File: src/main/java/com/ifmo/epampractice/dao/PassportDao.java (hunk 1 above)
Comment (ChirkinaTanya) [reply src/main/java/com/ifmo/epampractice/dao/PassportDao.java]: fixed

**Review by a-deriabin — APPROVED** [review]

**Review by PavelEzhkov — APPROVED** [review]

**Pull request merged**